### ISSUE TITLE
Simple Sites - Site Editor keeps loading default template if it has a custom domain as primary

### ISSUE THREAD

**Issue by DavidSaladu** [Automattic/wp-calypso#77320]
### Quick summary

Seems identical to [this previous issue](https://github.com/Automattic/wp-calypso/issues/76105).

The user sees the loading icon indefinitely while trying to access the template editor through appearance->editor or directly through pages if the template is highlighted on top.

![Screen Capture on 2023-05-24 at 12-35-12](https://github.com/Automattic/wp-calypso/assets/79575219/f7491267-10a5-43c3-9f26-87f16b8b5b97)

### Steps to reproduce

Access site
access appearance -> editor

### What you expected to happen

The default template should load in the background and clicking there should open it.

### What actually happened

a loading icon stays in the background indefinitely.

(A workaround can be applied by clicking on the templates section on the left sidebar, and accessing the template through there.)

### Impact

Most (> 50%)

### Available workarounds?

Yes, easy to implement

### Logs or notes

- MacOs Ventura 13.4
- Chrome Version 113.0.5672.126 (Official Build) (x86_64)
- Theme rainfall and Archeo

**Comment by DavidSaladu** [comment]
6303284-zd-woothemes
6323868-zd-woothemes

**Comment by daria2303** [comment]
HCs reported these: 
6320672-zd-woothemes
6322842-zd-woothemes
6323254-zd-woothemes

**Comment by kingsleyinfo** [comment]
Another one here:
6323933-zd-woothemes

**Comment by kylemcph** [comment]
Another report here: 
6324038-zd-woothemes

**Comment by kylemcph** [comment]
Another report here: 

6324160-zd-woothemes

**Comment by csabarakasz** [comment]
6324175-zd-woothemes

**Comment by philnick206** [comment]
Reported here: 6324157-zd-woothemes

**Comment by david-binda** [comment]
A fix for the issue, which was really only affecting sites running the Gutenberg v15.8.1, has been deployed via D111753-code

**Comment by jp-imagines** [comment]
I just got another report in 6314969-zd-woothemes, and the workaround (selecting a template from the list) does **not** work. Selecting a template, page, or template part still results in the infinite load behavior, so the Site Editor is completely inaccessible.

@david-binda has that fix been released to all Simple sites?

Edit: this was addressed as a separate issue at p1684941204447469/1684925111.213389-slack-C02FMH4G8. The above user has already been updated that things are fixed; they _don't_ need another follow-up.
